Sources/Internal/GrpcClient.swift

```diff
@@ -195,22 +193,20 @@ actor GrpcClient: CustomStringConvertible {
     }
 
     // Add AppCheck token if available
-    if appCheckEnabled {
-      do {
-        if let token = try await appCheck?.token(forcingRefresh: false) {
-          headers.add(name: RequestHeaders.appCheckHeader, value: "\(token)")
-          FirebaseLogger.dataConnect
-            .debug("App Check token added: \(token)")
-        } else {
-          FirebaseLogger.dataConnect
-            .debug("App Check token unavailable. Not adding App Check header.")
-        }
-      } catch {
+    do {
+      if let token = try await appCheck?.token(forcingRefresh: false) {
```

Comment:
Did you test a scenario where app check isn't enabled? Basically confirm that this call doesn't block and fails immediately if app check isn't enabled.

Reply:
LGTM. Feel free to merge the CI failures are due to a networking issue.

Reply:
Could you please approve this PR to enable merging?